src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -131,41 +142,61 @@ export class DomEventPlugin implements Plugin {
         if (domEvent.cssLocator) {
-            const elementList = document.querySelectorAll(domEvent.cssLocator);
-            elementList.forEach((element: HTMLElement) => {
-                element.addEventListener(eventType, eventListener);
+            const cssLocatedElements = document.querySelectorAll(
+                domEvent.cssLocator
+            );
+            cssLocatedElements.forEach((element) => {
+                identifiedElementList.push(element);
             });
         } else if (domEvent.elementId) {
-            document
-                .getElementById(domEvent.elementId)
-                ?.addEventListener(eventType, eventListener);
+            const identifiedElement = document.getElementById(
+                domEvent.elementId
+            );
+            if (identifiedElement) {
+                identifiedElementList.push(identifiedElement);
+            }
         } else if (domEvent.element) {
-            domEvent.element.addEventListener(eventType, eventListener);
+            identifiedElementList.push(domEvent.element);
         }
+
+        identifiedElementList.forEach((element) => {
+            element.addEventListener(eventType, eventListener);
+            elementEventListenerList.push({ element, eventListener });
+        });
+
+        this.eventListenerMap.set(domEvent, elementEventListenerList);
     }
 
     private removeEventHandler(domEvent: TargetDomEvent): void {
-        const eventListener:
-            | EventListener
-            | undefined = this.eventListenerMap.get(domEvent);
-
-        if (domEvent.cssLocator && eventListener) {
-            const elementList = document.querySelectorAll(domEvent.cssLocator);
-            elementList.forEach((element: HTMLElement) => {
+        const elementEventListenerList = this.eventListenerMap.get(domEvent);
+        if (elementEventListenerList) {
+            elementEventListenerList.forEach((elementEventListener) => {
+                const element = elementEventListener.element;
+                const eventListener = elementEventListener.eventListener;
                 element.removeEventListener(domEvent.event, eventListener);
             });
-        } else if (domEvent.elementId && eventListener) {
-            const element = document.getElementById(domEvent.elementId);
-            if (element) {
-                element.removeEventListener(domEvent.event, eventListener);
-            }
-        } else if (domEvent.element && eventListener) {
-            domEvent.element.removeEventListener(domEvent.event, eventListener);
+            this.eventListenerMap.delete(domEvent);
         }
-        this.eventListenerMap.delete(domEvent);
+    }
+
+    private observeDOMMutation() {
+        this.observer = new MutationObserver((mutations) => {
+            mutations.forEach(() => {
+                this.removeListeners();
+                this.addListeners();
+            });
```

Comment:
Isn't the remove/add operation idempotent?
```suggestion
            this.removeListeners();
            this.addListeners();
```

Reply:
Refactored